com.unity.render-pipelines.high-definition/Runtime/Lighting/LightCookieManager.cs

```diff
@@ -268,17 +412,135 @@ public void ReserveSpace(Texture cookie)
+
+            int projectionSize  = 2*cookie.width;
+
+            if (projectionSize < k_MinCookieSize)
+                return;
+
+            if (!m_CookieAtlas.ReserveSpace(m_CookieAtlas.GetTextureID(cookie), projectionSize, projectionSize))
+                m_2DCookieAtlasNeedsLayouting = true;
+        }
+
+        public void ReserveSpaceCube(Texture cookieA, Texture cookieB)
+        {
+            if (cookieA == null && cookieB == null)
+                return;
+
+            Debug.Assert(cookieA.dimension == TextureDimension.Cube && cookieB.dimension == TextureDimension.Cube);
+
+            int projectionSize  = 2*(int)Mathf.Max(cookieA.width, cookieB.width);
+
+            if (projectionSize < k_MinCookieSize)
+                return;
+
+            if (!m_CookieAtlas.ReserveSpace(m_CookieAtlas.GetTextureID(cookieA, cookieB), projectionSize, projectionSize))
+                m_2DCookieAtlasNeedsLayouting = true;
+        }
+
+        private RTHandle ProjectCubeTo2D(CommandBuffer cmd, Texture cookie, int projectionSize)
+        {
+            Debug.Assert(cookie.dimension == TextureDimension.Cube);
+
+            RTHandle projected = RTHandles.Alloc(
+                                        projectionSize,
+                                        projectionSize,
+                                        colorFormat: cookieFormat,
+                                        enableRandomWrite: true);
+            int usedKernel = m_KernalFastOctahedral;
+
+            float invSize = 1.0f/((float)projectionSize);
+
+            cmd.SetComputeTextureParam(m_ProjectCubeTo2D, usedKernel, HDShaderIDs._Input,  cookie);
+            cmd.SetComputeTextureParam(m_ProjectCubeTo2D, usedKernel, HDShaderIDs._Output, projected);
+            cmd.SetComputeFloatParams (m_ProjectCubeTo2D,             HDShaderIDs._ScaleBias,
+                                        invSize, invSize, 0.5f*invSize, 0.5f*invSize);
+
+            const int groupSizeX = 8;
+            const int groupSizeY = 8;
+            int threadGroupX = (projectionSize + (groupSizeX - 1))/groupSizeX;
+            int threadGroupY = (projectionSize + (groupSizeY - 1))/groupSizeY;
+
+            cmd.DispatchCompute(m_ProjectCubeTo2D, usedKernel, threadGroupX, threadGroupY, 1);
+
+            return projected;
+        }
+
+        public Vector4 FetchCubeCookie(CommandBuffer cmd, Texture cookie)
+        {
+            Debug.Assert(cookie != null);
+            Debug.Assert(cookie.dimension == TextureDimension.Cube);
+
+            int projectionSize = 2*(int)Mathf.Max((float)m_CookieCubeResolution, (float)cookie.width);
+            if (projectionSize < k_MinCookieSize)
+                return Vector4.zero;
+
+            if (!m_CookieAtlas.IsCached(out var scaleBias, cookie) && !m_NoMoreSpace)
+                Debug.LogError($"Cube cookie texture {cookie} can't be fetched without having reserved. You can try to increase the cookie atlas resolution in the HDRP settings.");
```

Comment:
@JordanL8

Reply:
$"Unity cannot fetch the Cube cookie texture: {cookie} because it is not on the cookie atlas. To resolve this, open your HDRP Asset and increase the resolution of the cookie atlas."